treeherder/perf/alerts.py

```diff
@@ -120,3 +122,98 @@ def generate_new_alerts_in_series(signature):
                         'new_value': new_value,
                         't_value': t_value
                     })
+
+
+@dataclass
+class AlertSelectionConfig():
+    max_alerts_selected = 5
+    platforms_of_interest = ('windows10', 'windows7', 'linux', 'osx', 'android')
+
+
+def os_relevance(alert_platform: str):
+    platforms = [
+        {
+            'name': 'windows10',
+            'relevance': 5
+        },
+        {
+            'name': 'windows7',
+            'relevance': 4
+        },
+        {
+            'name': 'linux',
+            'relevance': 3
+        },
+        {
+            'name': 'osx',
+            'relevance': 2
+        },
+        {
+            'name': 'android',
+            'relevance': 1
+        }
+    ]
+
+    for platform in platforms:
+        if platform['name'] in alert_platform:
+            return platform['relevance']
+    return 0
+
+
+def ensure_platform_variety(sorted_all_alerts: list, alert_selection_config=AlertSelectionConfig):
+    # Note: Ensure that the sorted_all_alerts container has only
+    # platforms of interest (example: 'windows10', 'windows7', 'linux', 'osx', 'android').
+    # Otherwise, the below 'while' will loop forever.
+    # In order to achieve this, please filter sorted_all_alerts list with filter_alerts(alerts) before calling this function.
+    circular_list = cycle(alert_selection_config.platforms_of_interest)
+    platform_to_pick = next(circular_list)
+
+    picked_alerts = list()
+    current_alert = 0
+    while(current_alert < len(sorted_all_alerts)):
+        if platform_to_pick in sorted_all_alerts[current_alert].series_signature.platform.platform:
+            if len(picked_alerts) < alert_selection_config.max_alerts_selected:
+                picked_alerts.append(sorted_all_alerts[current_alert])
+            current_alert += 1
+        platform_to_pick = next(circular_list)
+    return picked_alerts
+
+
+def filter_alerts(alerts: list, alert_selection_config=AlertSelectionConfig):
+    # Be aware only of mentioned platforms of interest
+    filtered_alerts = list()
+    for alert in alerts:
+        for platform in alert_selection_config.platforms_of_interest:
+            if alert.series_signature.platform.platform.startswith(platform):
```

Comment:
These deeply nested attributes can be better written using this [SO suggestion](https://stackoverflow.com/questions/11975781/why-does-getattr-not-support-consecutive-attribute-retrievals).

Reply:
Not applicable anymore, because the function filter_alerts is replaced with a filter criteria method which is to be used as the first parameter to native filter().